### PR TITLE
feat(core): dot core explain tx

### PR DESCRIPTION
**Pull request by nimraarif21**

### CODE REVIEW

File: modules/core/src/v2/coins/dot.ts [hunk 1]
```diff
@@ -221,6 +210,27 @@ export class Dot extends BaseCoin {
...
   }
 
+  /**
+   * Explain/parse transaction
+   * @param params
+   * @param callback
+  */
+  async explainTransaction(params: ExplainTransactionOptions): Promise<accountLib.Dot.TransactionExplanation> {
+    const txHex = params.txPrebuild.txHex;
+    if (!txHex || !params.txPrebuild.validity || !params.txPrebuild.key || !params.txPrebuild.referenceBlock || !params.txPrebuild.version) {
```
Comment (aaldemir) [inline modules/core/src/v2/coins/dot.ts:220]: `!params.txPrebuild.version` is it possible for version to ever be `0`?

File: modules/core/src/v2/coins/dot.ts [hunk 2]
```diff
@@ -221,6 +210,27 @@ export class Dot extends BaseCoin {
     return { txHex: signedTxHex };
   }
 
+  /**
+   * Explain/parse transaction
+   * @param params
+   * @param callback
+  */
+  async explainTransaction(params: ExplainTransactionOptions): Promise<accountLib.Dot.TransactionExplanation> {
+    const txHex = params.txPrebuild.txHex;
```
Comment (aaldemir) [inline modules/core/src/v2/coins/dot.ts:219]: there might be an issue here if `params.txPrebuild` is undefined - maybe move this below the invalid params check?

File: modules/core/src/v2/coins/dot.ts [hunk 3]
```diff
@@ -221,6 +210,27 @@ export class Dot extends BaseCoin {
...
+      throw new Error('missing explain tx parameters');
+    }
+    const sender = new accountLib.Dot.KeyPair({ pub: params.txPrebuild.key }).getAddress();
+    const factory = accountLib.register(this.getChain(), accountLib.Dot.TransactionBuilderFactory);
+    const txBuilder = factory.from(txHex);
+    txBuilder
+      .validity(params.txPrebuild.validity)
+      .referenceBlock(params.txPrebuild.referenceBlock)
+      .sender({ address: sender })
+      .version(params.txPrebuild.version);
```
Comment (argjv) [inline modules/core/src/v2/coins/dot.ts:230]: Aren't these in the txHex already?

File: modules/core/src/v2/coins/dot.ts (hunk 1 above)
Comment (nimraarif21) [reply modules/core/src/v2/coins/dot.ts:220]: No it cannot be. Version is always 1 or greater.

File: modules/core/src/v2/coins/dot.ts (hunk 3 above)
Comment (nimraarif21) [reply modules/core/src/v2/coins/dot.ts:230]: No they have to be set beforehand. They cannot be decoded from the hex.

**Review by aaldemir — APPROVED** [review]
LGTM

File: modules/core/src/v2/coins/dot.ts [hunk 4]
```diff
@@ -221,6 +210,27 @@ export class Dot extends BaseCoin {
...
+   * @param params
+   * @param callback
+  */
+  async explainTransaction(params: ExplainTransactionOptions): Promise<accountLib.Dot.TransactionExplanation> {
+    if (!params.txPrebuild.txHex || !params.txPrebuild.validity || !params.txPrebuild.key || !params.txPrebuild.referenceBlock || !params.txPrebuild.version) {
+      throw new Error('missing explain tx parameters');
+    }
+    const txHex = params.txPrebuild.txHex;
+    const sender = new accountLib.Dot.KeyPair({ pub: params.txPrebuild.key }).getAddress();
+    const factory = accountLib.register(this.getChain(), accountLib.Dot.TransactionBuilderFactory);
```
Comment (0xabrarhussain) [inline modules/core/src/v2/coins/dot.ts:224]: I believe DOT also has tokens right? might need a call for `getBaseChain` when `dotTokens` is added in

**Pull request closed without merging**